projects/dedust/index.js

```diff
@@ -0,0 +1,22 @@
+const { get } = require('../helper/http')
+const { transformDexBalances } = require('../helper/portedTokens')
+
+module.exports = {
+  misrepresentedTokens: true,
+  timetravel: false,
+  ton: {
+    tvl: async () => {
+      const pools = await get('https://api.dedust.io/v1/pools')
```

Comment:
Is there some way to fetch this info on chain?

Reply:
Yes, it's possible, but it's kind of cumbersome.

Each pool is a separate smart contract, it's responsible for calculations but all tokens are managed by next contract (it'll be changed, so it's better to have such kind of abstraction as an API layer):

https://tonapi.io/account/EQCcv7J5FyDI3SaqgFv4q1GCxbtoGZr3umefmvewNS7Xlukw

Here we can see reserves of pair SCALE / TON (2nd and 3rd numbers):
https://mainnet-v4.tonhubapi.com/block/27680435/EQA1ZM9KUwGz0APPLyOtsUlMWzcivIzMGSOlT1toiD-Cbrk3/run/get_balances

`34451813129701` (34451 TON) + `651262317325571` (651262 SCALE) = $173.6K TVL ( SCALE / TON )

This contract holds all wrapped TONs (only part of them are in liquidity pools, but majority):

https://tonapi.io/account/EQCPPAuDDsUOFjXC7FbMD2d024NhwM7d5CpsWerXQ4qLoL9w